### DISCUSSION TITLE
Support for node order discontiguous?

### DISCUSSION THREAD

**Discussion by mgao97** [pyg-team/pytorch_geometric#6816] [Q&A]
Hi!, I have one question about the node order within the graph data object.

Does pyg support node order discontiguous?

When I try to gat node indexes from edge indexs, I run into the following problem:
"index xxx out of bounds for dimension ..."

**Comment by rusty1s** [comment]
PyG requires that node indices range from `0` to `num_nodes - 1`. This is needed to achieve an efficient memory access pattern.

**Reply by mgao97** [reply]
Got it, thanks a lot!
